src/IslandoraNodeDeleteSubscriber.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Drupal\islandora_entity_status;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\node\NodeInterface;
+
+/**
+ * Subscriber to handle deletion of related Embargoes when a node is deleted.
+ */
+class IslandoraNodeDeleteSubscriber {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new CustomEntityDeleter object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * Deletes the associated embargos.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The Islandora object node being deleted.
+   */
+  public function deleteAssociatedCustomEntity(NodeInterface $node): void {
+    $custom_entity_storage = $this->entityTypeManager->getStorage('embargo');
```

Comment:
This would effectively be creating a dependency on the `embargo` module, yeah? Probably not desired.

Reply:
@adam-vessey please suggest in which module I can implement this. https://github.com/discoverygarden/embargo module?

Reply:
Would be more of the target, yes. Still not convinced that such functionality is desired.

Like, something like a scan to find occurrences of `::getEmbargoedNode()` that don't account for it being able to be `NULL` might be more in order?

The ticket is more about error suppression than data maintenance that this PR is getting into, or so I read it. As in, much like https://github.com/discoverygarden/embargo/pull/41 , might also need to hit: https://github.com/discoverygarden/embargo/blob/a075aa4a994472666adf9cdd1af48891fb675323/src/Plugin/search_api/processor/EmbargoJoinProcessor.php#L230